### PR TITLE
[ARVIR2] Reset AR world 📱

### PR DESCRIPTION
**Pull request by alloy**
Closes https://artsyproduct.atlassian.net/browse/EV-148

This should thus far have the design up to spec.

### CODE REVIEW

File: Artsy/View_Controllers/ARVIR/AR/ARVIRHorizontalPlaneInteractionController.m [hunk 1]
```diff
@@ -179,7 +179,8 @@ - (void)placeArtwork
     }
 }
 
-
+// TODO: Figure out in how far this is done yet?
+//       And why does the ARSession need to be re-created?
 - (void)restart
 {
     [self.ghostWallLine removeFromParentNode];
```
Comment (alloy) [inline Artsy/View_Controllers/ARVIR/AR/ARVIRHorizontalPlaneInteractionController.m]: @orta Was this an approach that wasn’t working or was it partially done?

File: Artsy/View_Controllers/ARVIR/VCs/ARAugmentedFloorBasedVIRViewController.m [hunk 2]
```diff
@@ -22,13 +22,34 @@
...
+@end
+
+@interface _ARClearFlatButton : ARClearFlatButton
+@end
+@implementation _ARClearFlatButton
+- (void)setHighlighted:(BOOL)highlighted;
+{
+    // no-op
+}
+@end
```
Comment (alloy) [inline Artsy/View_Controllers/ARVIR/VCs/ARAugmentedFloorBasedVIRViewController.m]: These should either be renamed to be more unique or we should add a flag to disable highlights to Artsy-UIButtons. I’m on the fence, what say you?

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/View_Controllers/ARVIR/AR/ARVIRHorizontalPlaneInteractionController.m (hunk 1 above)
Comment (orta) [reply Artsy/View_Controllers/ARVIR/AR/ARVIRHorizontalPlaneInteractionController.m]: ( for others reading along, this was copy pasta from the other ARVIR )

File: Artsy/View_Controllers/ARVIR/VCs/ARAugmentedFloorBasedVIRViewController.m (hunk 2 above)
Comment (orta) [reply Artsy/View_Controllers/ARVIR/VCs/ARAugmentedFloorBasedVIRViewController.m]: I think this is fine, if it's ever needed in more places then we can abstract better

**Comment by alloy** [comment]
Ok this is ready to go 👍

File: Artsy/View_Controllers/ARVIR/AR/ARVIRHorizontalPlaneInteractionController.m [hunk 3]
```diff
@@ -53,21 +53,46 @@ @implementation ARVIRHorizontalPlaneInteractionController
...
+    self.wall = nil;
+
+    [self.artwork removeFromParentNode];
+    self.artwork = nil;
+
+    [self.ghostArtwork removeFromParentNode];
+    self.ghostArtwork = nil;
+
+    [self.session runWithConfiguration:self.session.configuration
+                               options:ARSessionRunOptionResetTracking | ARSessionRunOptionRemoveExistingAnchors];
```
Comment (orta) [inline Artsy/View_Controllers/ARVIR/AR/ARVIRHorizontalPlaneInteractionController.m]: ah, mutable state, sure

File: Artsy/View_Controllers/ARVIR/VCs/ARAugmentedFloorBasedVIRViewController.m [hunk 4]
```diff
@@ -168,25 +225,48 @@ - (void)viewDidLoad
...
-    [self.view addSubview:informationView];
-    [informationView alignLeading:@"0" trailing:@"0" toView:self.view];
-    [informationView constrainHeight:@"180"];
+    UIView *informational = self.informationView;
+    if (informational.alpha > 0) {
+        return;
+    }
+    [UIView animateIf:animated
+             duration:ARInformationalViewDisplayAnimationDuration
+              options:UIViewAnimationOptionCurveEaseOut
```
Comment (orta) [inline Artsy/View_Controllers/ARVIR/VCs/ARAugmentedFloorBasedVIRViewController.m]: nice move

**Comment by orta** [comment]
Yeah, this looks good to me, gonna ship a beta

**Pull request merged**